xmtp/src/client.rs

```diff
@@ -11,16 +13,28 @@ pub enum Network {
     Prod,
 }
 
-pub struct Client<P>
+#[derive(Debug, Error)]
+pub enum ClientError {
+    #[error("unknown client error")]
+    Unknown,
+}
+
+pub struct Client<P, S>
 where
     P: Persistence,
 {
     pub network: Network,
     pub persistence: NamespacedPersistence<P>,
     pub account: VmacAccount,
+    pub(super) _store: S,
 }
 
-impl<P: Persistence> Client<P> {
+impl<P, S> Client<P, S> where P: Persistence {}
+
+impl<P, S> Client<P, S>
+where
+    P: Persistence,
+{
```

Comment:
The generic type S is specifically unbounded. Type checking doesn't occur on the StorageMgr during initialization. Instead it occurs when the object is stored. This is still compile time checked however each Object,StorageMgr is free to determine what the appropriate interface is, which minimizes impacts of adding types/changing implementations.

Reply:
I updated my above comment as I wasn't making much sense. 

Any struct can store persist an object by calling `object.store(&storage)`. In the case of the client it does not need to know anything about the storage client to persist messages. To my knowledge the Client does not need to fetch messages -- consumers can set their own fetch interface ( Next PR)